File: src/BaseHmiTypes/Projects/HmiProjectItemDescriptor.cs
namespace BaseHmiTypes.Projects;

public class HmiProjectItemDescriptor : IHmiProjectItemDescriptor
{
    public string Id { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Path { get; set; } = string.Empty;

    public HmiProjectItemKind Kind { get; set; } = HmiProjectItemKind.Unknown;

    public HmiProjectFolderType FolderType { get; set; } = HmiProjectFolderType.Unknown;

    public string? SourceType { get; set; }
}

File: typescript/base-hmi-types/src/projects/HmiProjectItemDescriptor.ts
import { HmiProjectFolderType } from "./HmiProjectFolderType.js";
import { HmiProjectItemKind } from "./HmiProjectItemKind.js";

export interface IHmiProjectItemDescriptor {
  id: string;
  name: string;
  path: string;
  kind: HmiProjectItemKind;
  folderType: HmiProjectFolderType;
  sourceType?: string | undefined;
}

export class HmiProjectItemDescriptor implements IHmiProjectItemDescriptor {
  id = "";
  name = "";
  path = "";
  kind: HmiProjectItemKind = HmiProjectItemKind.Unknown;
  folderType: HmiProjectFolderType = HmiProjectFolderType.Unknown;
  sourceType?: string | undefined;
}
